pytest/tests/sanity/rpc_tx_resharding.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+"""
+Testing RPC call to transaction status after a resharding event.
+We create two account that we know would fall into different shards after resharding.
+We submit a transfer transaction between the accounts and verify the transaction status after resharding.
+"""
+
+import sys
+import unittest
+import pathlib
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+from configured_logger import logger
+import cluster
+import resharding_lib
+import transaction
+from utils import MetricsTracker, poll_blocks
+import key
+
+STARTING_AMOUNT = 123 * (10**24)
+
+
+class ReshardingTest2(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.epoch_length = 5
+        self.config = cluster.load_config()
+        self.binary_protocol_version = cluster.get_binary_protocol_version(
+            self.config)
+        assert self.binary_protocol_version is not None
+
+        self.target_shard_layout_version = resharding_lib.get_target_shard_layout_version(
+            self.binary_protocol_version)
+        self.target_num_shards = resharding_lib.get_target_num_shards(
+            self.binary_protocol_version)
+
+    def __setup_account(self, account_id, nonce):
+        encoded_block_hash = self.node.get_latest_block().hash_bytes
+        account = key.Key.from_random(account_id)
```

Comment:
What does this do?

Reply:
Creates two different accounts with some specific balance in shard3 and (future) shard4